core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -422,25 +430,31 @@ class ContainerPool(
 
     /* A background thread that
      *   1. Kills leftover action containers on startup
-     *   2. Periodically re-populates the container pool with fresh (un-instantiated) nodejs containers.
+     *   2. (actually a Future) Periodically re-populates the container pool with fresh (un-instantiated) nodejs containers.
      *   3. Periodically tears down containers that have logically been removed from the system
-     *   4. Each of the significant method subcalls are gaurded to not throw an exception.
+     *   4. Each of the significant method subcalls are guarded to not throw an exception.
      */
     private def nannyThread(allContainers: Seq[ContainerState]) = new Thread {
         override def run {
-            implicit val tid = TransactionId.invokerWarmup
+            implicit val tid = TransactionId.invokerNanny
             if (!standalone) {
                 killStragglers(allContainers)
-            }
-            while (true) {
-                Thread.sleep(100) // serves to prevent busy looping
-                // create a new stem cell if the number of warm containers is less than the count allowed
+                // Create a new stem cell if the number of warm containers is less than the count allowed
                 // as long as there is slack so that any actions that may be waiting to create a container
                 // are not held back; Note since this method is not fully synchronized, it is possible to
                 // start this operation while there is slack and end up waiting on the docker lock later
-                if (!standalone && getNumberOfIdleContainers(warmNodejsKey) < WARM_NODEJS_CONTAINERS && slack() > 0) {
-                    addWarmNodejsContainer()(tid)
+                val warmupInterval = 100.milliseconds
+                Scheduler.scheduleWaitAtLeast(warmupInterval) { () =>
+                    implicit val tid = TransactionId.invokerWarmup
+                    if (getNumberOfIdleContainers(warmNodejsKey) < WARM_NODEJS_CONTAINERS && slack() > 0 && toBeRemoved.size < RM_THRESHOLD) {
+                        addWarmNodejsContainer()(tid)
+                    } else {
+                        Future.successful(())
```

Comment:
We could but the `else` branch is not an unusual or error condition.  In fact, it is common and corresponds to when no one is using the warm containers and so any logging here would represent a ton of verbose logging for normal idle condition.